docs/backend_architecture/tasks.rst

```diff
@@ -38,22 +38,25 @@ We will refer to below sample code in the later sections also.
 
 .. code-block:: python
 
+    from rest_framework import serializers
+
     from kolibri.core.tasks.decorators import register_task
     from kolibri.core.tasks.job import Priority
-    from kolibri.core.device.permissions import IsSuperuser
+    from kolibri.core.tasks.permissions import IsSuperAdmin
+    from kolibri.core.tasks.validation import JobValidator
 
-    def add_validator(req, req_data):
-        assert isinstance(req_data["a"], int)
-        assert isinstance(req_data["b"], int)
-        return {
-          "a": req_data["a"],
-          "b": req_data["b"],
-          "extra_metadata": {
-            "user": "kolibri"
-          }
-        }
+    class AddValidator(JobValidator):
+        a = serializers.IntegerField()
+        b = serializers.IntegerField()
 
-    @register_task(job_id="02", queue="maths", validator=add_validator, priority=Priority.HIGH, cancellable=False, track_progress=True, permission_classes=[IsSuperuser])
+        def validate(self, data):
+            if data['a'] + data['b'] > 100:
+                raise serializers.ValidationError("Sum of a and b should be less than 100")
+            job_data = super(AddValidator, self).validate(data)
+            job_data["extra_metadata"].update({"user": "kolibri"})
+            return job_data
+
+    @register_task(job_id="02", queue="maths", validator=add_validator, priority=Priority.HIGH, cancellable=False, track_progress=True, permission_classes=[IsSuperAdmin])
```

Comment:
`validator` should be `AddValidator`...?

Reply:
```suggestion
    @register_task(job_id="02", queue="maths", validator=AddValidator, priority=Priority.HIGH, cancellable=False, track_progress=True, permission_classes=[IsSuperAdmin])
```

Reply:
Yep that's what I meant there!